File: 2026POTD/FEBRUARY/12-02-2026/3713.cpp
class Solution {
public:
    int longestBalanced(string s) {
        
        int n = s.length();
        int maxLen = 0;

        for (int i = 0; i < n; i++) {

            vector<int> freq(26, 0);

            for (int j = i; j < n; j++) {

                freq[s[j] - 'a']++;

                int count = 0;
                int distinct = 0;
                bool balanced = true;

                for (int k = 0; k < 26; k++) {
                    if (freq[k] > 0) {
                        distinct++;

                        if (count == 0)
                            count = freq[k];
                        else if (freq[k] != count) {
                            balanced = false;
                            break;
                        }
                    }
                }

                if (balanced) {
                    int len = j - i + 1;
                    maxLen = max(maxLen, len);
                }
            }
        }

        return maxLen;
    }
};

File: 2026POTD/FEBRUARY/12-02-2026/3713.java
class Solution {
    public int longestBalanced(String s) {
        int max=0;
        for(int i=0;i<s.length();i++){
            Map<Character,Integer> map=new HashMap<>();
            for(int j=i;j<s.length();j++){
                char ch=s.charAt(j);
                map.put(ch,map.getOrDefault(ch,0)+1);
                if(balance(map)){
                max=Math.max(max,j-i+1);
            }
            }

        }
        return max;
    }
    public static boolean balance(Map<Character,Integer> freq){
        Set<Integer> set=new HashSet<>(freq.values());
        return set.size()==1;
    }
}  